src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -43,6 +46,51 @@ constexpr bool contains_indices_to_contract(
   }
 }
 
+/// \brief Given the list of the positions of the LHS tensor's spacetime indices
+/// where a generic spatial index is used and the list of positions where a
+/// concrete time index is used, determine whether or not the component at the
+/// given LHS multi-index should be computed
+///
+/// \details
+/// Not all of the LHS tensor's components may need to be computed. Cases when
+/// the component at a LHS multi-index should not be not evaluated:
+/// - If a generic spatial index is used for a spacetime index on the LHS,
+/// the components for which that index's concrete index is the time index
+/// should not be computed
+/// - If a concrete time index is used for a spacetime index on the LHS, the
+/// components for which that index's concrete index is a spatial index should
+/// not be computed
+///
+/// \param lhs_multi_index the multi-index of the LHS tensor to check
+/// \param lhs_spatial_spacetime_index_positions the positions of the LHS
+/// tensor's spacetime indices where a generic spatial index is used
+/// \param lhs_time_index_positions the positions of the LHS tensor's spacetime
+/// indices where a concrete time index is used
+/// \return Whether or not `lhs_multi_index` is a multi-index of a component of
+/// the LHS tensor that should be computed
+template <size_t NumLhsIndices, size_t NumLhsSpatialSpacetimeIndices,
+          size_t NumLhsConcreteTimeIndices>
+constexpr bool is_lhs_multi_index_to_evaluate(
```

Comment:
I'm really struggling to parse this function name. I don't think it makes grammatical sense (at least not to me). I think there's a verb missing.

Reply:
Ah, it's meant to read as, "Is this a LHS multi-index to evaluate?" / "Is this a LHS multi-index that should be evaluated?" Is `is_lhs_multi_index_evaluated` better?

Reply:
I'm actually leaning towards `is_lhs_multi_index_evaluated`, because it seems to simply answer "is this lhs multi-index one of the ones that is/will be evaluated?" in a generic sense, like a classification rather than it being specific to past/future tense, i.e. regardless of what point of the algorithm it is called in, it can tell us if this is a multi-index that we compute, whether we've already done it or not. I'll go with this for these reasons, but let me know if you feel otherwise, or if the naming still doesn't read well